src/main/java/br/com/apppersonal/apppersonal/service/PersonalService.java

```diff
@@ -0,0 +1,134 @@
+
+            UserEntity students = userRepository.findByIdAndRole(userId, Role.USER);
+
+            if (students == null) {
+                throw new UserNotFoundException("Usuário não encontrado");
+            }
+
+            students.setPersonalTrainerId(personalId); // Associa o usuário ao personal trainer
+            userRepository.save(students);
+
+            return ResponseEntity
+                    .status(HttpStatus.OK)
+                    .body(
+                            new ApiResponse(
+                                    true,
+                                    "Usuário associado com sucesso ao personal trainer"
+                            )
+                    );
+        } catch (Exception e) {
+            return ResponseEntity
+                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(
+                            new ApiResponse(
+                                    false,
+                                    e.getMessage()
+                            )
+                    );
+        }
+    }
+    public ResponseEntity<?> dissociateUserFromPersonal(Long userId) {
+        try {
+            if (userId == null) throw new ParameterNullException("Identificador do usuário não informado");
+
+            UserEntity students = userRepository.findByIdAndRole(userId, Role.USER);
+
+            if (students == null) {
+                throw new UserNotFoundException("Usuário não encontrado");
+            }
+
+            students.setPersonalTrainerId(null); // Remove a associação com o personal trainer
+            userRepository.save(students);
+
+            return ResponseEntity
+                    .status(HttpStatus.OK)
+                    .body(
+                            new ApiResponse(
+                                    true,
+                                    "Usuário dissociado com sucesso do personal trainer"
+                            )
+                    );
+        } catch (Exception e) {
+            return ResponseEntity
+                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(
+                            new ApiResponse(
+                                    false,
+                                    e.getMessage()
+                            )
+                    );
+        }
+    }
+
+    public ResponseEntity<?> listStudentsByPersonal(Long personalId) {
+        try {
+            if (personalId == null) throw new ParameterNullException("Identificador do personal trainer não informado");
+
+            // busca por todos users com role USER e que tenham o personal trainer id associado
+            List<UserEntity> students = userRepository.findByRoleAndPersonalTrainerId(Role.USER, personalId);
+
+            if (students.isEmpty()) {
+                throw new UserNotFoundException("Nenhum aluno encontrado");
+            }
```

Comment:
Ajusta if, colocar throw in line

```suggestion
            if (students.isEmpty()) throw new UserNotFoundException("Nenhum aluno encontrado");
```